### PR TITLE
Improved actors assignment to nodes

### PR DESCRIPTION
**Pull request by mjanczykowski**

### CODE REVIEW

File: common/src/main/scala/miss/supervisor/Supervisor.scala [hunk 1]
```diff
@@ -229,6 +249,25 @@ class Supervisor(config: Config) extends FSM[State, Data] {
...
+      .filter(nodeDims => cols % nodeDims._2 == 0 && nodeDims._2 / nodeDims._1 <= cols)
+    if (possibleRowSizes.isEmpty) {
+      return None
+    }
+
+    Some(possibleRowSizes.min(Ordering.by((dims: (Int, Int)) => dims._1 + dims._2)))
+  }
+
+  private def factors(num: Int) = {
+    (2 to num).filter(divisor =>
```
Comment (salceson) [inline common/src/main/scala/miss/supervisor/Supervisor.scala]:
```scala
(2 to math.sqrt(num).toInt)
.filter(divisor => num % divisor == 0)
.flatMap(divisor => Seq(divisor, num / divisor)
```
should be more efficient.

File: common/src/main/scala/miss/supervisor/Supervisor.scala [hunk 2]
```diff
@@ -229,6 +249,25 @@ class Supervisor(config: Config) extends FSM[State, Data] {
...
+
+    val possibleRowSizes = rowsFactors.filter(factor => workerCores % factor == 0)
+      .map(f => (f, workerCores / f))
+      .seq
+      .filter(nodeDims => cols % nodeDims._2 == 0 && nodeDims._2 / nodeDims._1 <= cols)
+    if (possibleRowSizes.isEmpty) {
+      return None
+    }
+
+    Some(possibleRowSizes.min(Ordering.by((dims: (Int, Int)) => dims._1 + dims._2)))
```
Comment (salceson) [inline common/src/main/scala/miss/supervisor/Supervisor.scala]: 259 - 263 could be:
```scala
possibleRowSizes sortWith {
  case ((rows1, cols1), (rows2, cols2)) => rows1 + cols1 < rows2 + cols2
} headOption
```

**Review by salceson — APPROVED** [review]
Good work, but some code can be better :) :+1:

File: common/src/main/scala/miss/supervisor/Supervisor.scala (hunk 2 above)
Comment (salceson) [reply common/src/main/scala/miss/supervisor/Supervisor.scala]: or even:
```scala
possibleRowSizes.sortBy(nodeDims => nodeDims._1 + nodeDims._2).headOption
```

File: common/src/main/scala/miss/supervisor/Supervisor.scala (hunk 1 above)
Comment (mjanczykowski) [reply common/src/main/scala/miss/supervisor/Supervisor.scala]: I've added 1 and num too

Comment (salceson) [reply common/src/main/scala/miss/supervisor/Supervisor.scala]: Hmm... why?

**Pull request merged**